### ISSUE TITLE
Scan data in run issue

### ISSUE THREAD

**Issue by raquelalvarezbanos** [mantidproject/mantid#18714]
This was reported by Jos:

> Please find attached a script to load two runs we had this cycle which appear to have scan data in the loaded data which can only be from negative run time. I don’t know why this happens, or if it happens every time, but it is pretty weird. The only thing I can think is that the nexus contains data from before the run started (which is fine as far as I understand), but that the t=0 (start of the run) is somehow not taken account of when loading the file? Do you have any thoughts on this Raquel? It has happened twice this cycle, and we aren’t sure if it always happens or just we somehow got something weird going on.

The script is (set the instrument to `OFFSPEC`): 

```
LoadISISNexus("40708",OutputWorkspace="single_period")
LoadEventNexus("40708",OutputWorkspace="single_period_t=0_onwards",FilterByTimeStart=0)
Rebin("single_period_t=0_onwards",[0,20,100000],OutputWorkspace="single_period_t=0_onwards")


LoadISISNexus("40648",OutputWorkspace="two_period")
LoadEventNexus("40648",OutputWorkspace="two_period_t=0_onwards",FilterByTimeStart=0)
Rebin("two_period_t=0_onwards",[0,20,100000],OutputWorkspace="two_period_t=0_onwards")
```

Workspaces `two_period` and `two_period_t=0_onwards` should be the same, but they are not.

I looked into this issue and I don't think there is a problem with `FilterByTimeStart=0` not being taken into account. If you set for instance `FilterByTimeStart=100` the output workspace has a smaller number of events, which indicates that the start time was taken into account. I think it could be a problem in those nexus files, and scientists were going to ask Freddie about this. This issue is to ask Jos if they have been able to clarify with Freddie what was happening.

**Comment by raquelalvarezbanos** [comment]
I am pasting below an email I sent Jos about this issue

**Comment by raquelalvarezbanos** [comment]
Hi Jos,

I was expecting that, if there were scan data from negative times, running:

test_1 = LoadEventNexus("40708", FilterByTimeStart=0)
test_2 = LoadEventNexus("40708")

would give two different workspaces, but apparently they are the same.

But going back to your problem, the issue was that, as shown in your script, when comparing the event workspace(i.e. the workspace loaded with “LoadEventNexus”) to the histogram workspace (i.e. the workspace loaded with “LoadISISNexus”), they are different. I’ve been looking into the nexus file and the loading algorithms and I think Mantid is just loading what’s in the nexus file. This is what I found when looking into this:

-	 “LoadISISNexus” essentially opens entry “raw_data_1/detector_1/counts” (see figure below). This is what gets loaded into the spectra in the loaded workspace.
-	 “LoadEventNexus” reads various things in “raw_data_1/detector_1_events/” (event_id, event_index, event_time_offset, event_time_zero, and total_counts).

![scan-data-issue-fig1](https://cloud.githubusercontent.com/assets/8956985/22647684/e292d264-ec6a-11e6-98f2-ae6bbc8c2dc8.png)

-	I think Mantid is just loading what’s in the nexus file. This is how I’ve checked this. Below is the color fill plot you showed me the other day in the meeting, and as we saw there are some high intensity peaks in the `single_period` workspace:

![scan-data-issue-fig2](https://cloud.githubusercontent.com/assets/8956985/22647725/1fd123f6-ec6b-11e6-904d-e4988da88777.jpg)

In the figure below I am just focusing on spectrum with id 403. This spectrum has a high intensity peak at TOF ~ 97600. Specifically, bin 4876 has 13,819 counts:

![scan-data-issue-fig3](https://cloud.githubusercontent.com/assets/8956985/22647778/542c8280-ec6b-11e6-8622-11ab7a926eb0.png)

This number, 13819 counts, is exactly what I find if I inspect the nexus file using HDFView (note that in the plot on the right, HDView is plotting the counts as a function of the detector ID for bin 4876):

![scan-data-issue-fig4](https://cloud.githubusercontent.com/assets/8956985/22647812/7067775c-ec6b-11e6-95c6-428d0ffc65a1.png)

In summary, those peaks we are seeing in the left color fill plot seem to be in the nexus file. However, when we load the run as an event run, it seems that these counts are not recorded in `detector_1_events`, I am not sure why. I am sorry I am not being more helpful, let me know if you can think of anything else I can check for you.

Raquel

**Comment by gemmaguest** [comment]
Closing as stale because we are unlikely to get around to this any time soon
